pkg_resources/__init__.py

```diff
@@ -804,9 +806,14 @@ def resolve(self, requirements, env=None, installer=None,
                 continue
             # If the req has a marker, evaluate it -- skipping the req if
             # it evaluates to False.
-            # https://github.com/pypa/setuptools/issues/523
-            _issue_523_bypass = True
-            if not _issue_523_bypass and req.marker and not req.marker.evaluate():
+            if req.marker:
+                result = []
+                if req in extra_req_mapping:
+                    for extra in extra_req_mapping[req] or ['']:
```

Comment:
The tests do pass, but it actually ends up not calling marker.evaluate() in that case at all. The only thing I could think of to deal with that was asserting len(result) >= 1, but that concerned me that it might cause some head-scratching further down the line.

I'll see if I can come up with a test case.